pkg/storage/snapshot/restore.go

```diff
@@ -1005,17 +1037,22 @@ func (ctrl *VMRestoreController) getSnapshotContent(vmRestore *snapshotv1.Virtua
 		return nil, fmt.Errorf("VMSnapshot %s does not exist", objKey)
 	}
 
-	vms := obj.(*snapshotv1.VirtualMachineSnapshot).DeepCopy()
-	if !VmSnapshotReady(vms) {
+	vmSnapshot := obj.(*snapshotv1.VirtualMachineSnapshot).DeepCopy()
+	if vmSnapshotFailed(vmSnapshot) {
+		return nil, fmt.Errorf("VMSnapshot %s failed and is invalid to use", objKey)
+	} else if !VmSnapshotReady(vmSnapshot) {
 		return nil, fmt.Errorf("VMSnapshot %s not ready", objKey)
 	}
 
-	if vms.Status.VirtualMachineSnapshotContentName == nil {
+	if vmSnapshot.Status.VirtualMachineSnapshotContentName == nil {
 		return nil, fmt.Errorf("no snapshot content name in %s", objKey)
 	}
+	return vmSnapshot, nil
```

Comment:
nit: replace all the ifs with a switch

Reply:
the ifs are not checking the same field so I dont see how it can be done neatly

Reply:
you can group it together since other otherwise you are always returning
```golang
switch {
     case vmSnapshotFailed(vmSnapshot):
         return nil, fmt.Errorf("VMSnapshot %s failed and is invalid to use", objKey)
     case !VmSnapshotReady(vmSnapshot):
         return nil, fmt.Errorf("VMSnapshot %s not ready", objKey)
     case vmSnapshot.Status.VirtualMachineSnapshotContentName == nil:
         return nil, fmt.Errorf("no snapshot content name in %s", objKey)
     default:
        return vmSnapshot, nil
}
```
It's just a suggestion

Reply:
ok